Implement feedback about financial tool

paulapreuss opened the pull request:
This PR is for implementing comments/fixing bugs based on feedback from the greenwerk

Bachibouzouk commented on app/cp_nigeria/helpers.py:
```diff
@@ -736,14 +736,17 @@ def collect_financial_params(self):
         )
         qs_ed = EconomicData.objects.filter(project=self.project).values("discount", "tax")
-        financial_params = pd.concat([pd.DataFrame.from_records(qs_eq), pd.DataFrame.from_records(qs_ed)], axis=1)
+        financial_params = qs_ed.first() | qs_eq.first()
```
A comment to what this command does would be welcome :)

paulapreuss replied:
That is the merge operator for dictionaries; it is basically doing the same as the deleted line (concatenating the EconomicData values with the EquityData values), but on a dictionary instead of a dataframe. I decided to change this because it was a bit silly to have a dataframe with only one row.

Bachibouzouk commented on app/cp_nigeria/helpers.py:
```diff
@@ -736,14 +736,17 @@ def collect_financial_params(self):
         )
         qs_ed = EconomicData.objects.filter(project=self.project).values("discount", "tax")
-        financial_params = pd.concat([pd.DataFrame.from_records(qs_eq), pd.DataFrame.from_records(qs_ed)], axis=1)
+        financial_params = qs_ed.first() | qs_eq.first()
```
A comment to what this command does would be welcome :)

paulapreuss commented on app/cp_nigeria/views.py:
```diff
@@ -844,10 +844,7 @@ def cpn_constraints(request, proj_id, step_id=STEP_MAPPING["economic_params"]):
-            equity_data.debt_start = scenario.start_date.year
-            equity_data.scenario = scenario
-            equity_data.save()
+            equity_data = equity_form.save()
```
I deleted this code because now EquityData is already initialized on step 3 and the get_or_create method sets the debt_start and scenario as defined here, so it seemed duplicate

paulapreuss replied:
I undid this because it did break things as I suspected it might when I made this comment :')

paulapreuss commented:
This PR addresses all of the feedback given by the greenwerk in the _Economic Parameters_ e-mail, except for the custom community equity depending on the questionnaire answers (unless I missed something). I thought it was probably more effective for you to implement it than for me to first have to dive into the questionnaire structure. Can you implement this @Bachibouzouk ? 

I still haven't done super extensive validation with the new excel-files provided by Jannik, but I think it is more of a priority to have this PR merged soon-ish than it is to spend too much time looking for bugs (which should hopefully be pretty minor at this point)

Bachibouzouk commented on app/cp_nigeria/forms.py:
```diff
@@ -110,12 +109,12 @@ def __init__(self, *args, **kwargs):
-        ed = super().save(*args, **kwargs)
-        scenario = Scenario.objects.filter(project__economic_data=ed)
-        scenario.update(capex_fix=self.cleaned_data["capex_fix"])
+    # def save(self, *args, **kwargs):
+    #     ed = super().save(*args, **kwargs)
+    #     scenario = Scenario.objects.filter(project__economic_data=ed)
+    #     scenario.update(capex_fix=self.cleaned_data["capex_fix"])
```
Why was it commented out? If you don't need it we can simply delete it

Bachibouzouk reviewed (approved):
Thanks @paulapreuss  for implementing most of those changes!
